math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -3242,6 +3242,203 @@ RVec<typename RVec<T>::size_type> Enumerate(const RVec<T> &v)
    return ret;
 }
 
+/**
+ * @brief Produce RVec with N evenly-spaced entries from start to end inclusive.
+ *
+ * This function generates a vector of evenly spaced values, starting at @p start and ending at @p end,
+ * with exactly @p n elements. The spacing is computed as
+ * \f$\text{step} = \frac{\text{end} - \text{start}}{n-1}\f$, so that the vector includes both endpoints.
+ *
+ * The function is templated to allow for different arithmetic types. The deduced type @c Common_t is
+ * determined as follows: if the common type of @p T1, @p T2, and @p T3 is a floating point type, that type is used;
+ * otherwise, the arithmetic is performed using @c double.
+ *
+ * @tparam T1 Type of the start value. Default is double.
+ * @tparam T2 Type of the end value. Default is double.
+ * @tparam T3 Auxiliary type used for deducing the common type. Default is double.
+ * @tparam Common_t Deduced type used for arithmetic, which is std::common_type_t<T1, T2, T3> if that is a
+ * floating point type, or double otherwise.
+ *
+ * @param start The first value in the sequence.
+ * @param end The last value in the sequence.
+ * @param n The number of evenly spaced entries to produce (must be greater than 0).
+ *
+ * @return A vector (RVec<Common_t>) containing @p n evenly spaced values from @p start to @p end inclusive.
+ *
+ * @note If @p n is 1, the resulting vector will contain only the value @p start.
+ * @note The check `if (!n || (n > LLONG_MAX))` is used to ensure that:
+ *   - n is nonzero, and
+ *   - n does not exceed LLONG_MAX, which would indicate that a negative range (or other arithmetic issue)
+ *     has resulted in an extremely large unsigned value, thereby preventing an attempt to reserve an absurd
+ *     amount of memory.
+ *
+ * @par Example code, at the ROOT prompt:
+ * ~~~{.cpp}
+ * using namespace ROOT::VecOps;
+ * cout << LinSpace(-1, 5, 5) << "\n";
+ * // { -1, 0.5, 2, 3.5, 5 }
+ * cout << LinSpace(3, 12, 5) << "\n";
+ * // { 3, 5.25, 7.5, 9.75, 12 }
+ * cout << LinSpace(1.4, 13.66, 5) << "\n";
+ * // { 1.4, 4.465, 7.53, 10.595, 13.66 }
+ * ~~~
+ */
+template <typename T1 = double, typename T2 = double, typename T3 = double, typename Common_t = std::conditional_t<std::is_floating_point_v<std::common_type_t<T1, T2, T3>>, std::common_type_t<T1, T2, T3>, double>>
+inline RVec<Common_t> LinSpace(T1 start, T2 end, unsigned long long n = 128)
+{
+    RVec<Common_t> temp;
+    
+    if (!n || (n > LLONG_MAX)) // Check for invalid or absurd n.
```

Comment:
Thanks for the clarification in the comment. Can we perhaps use std::numeric_limits instead of LLONG_MAX?